counsel.el

```diff
@@ -6317,20 +6318,25 @@ list is passed to `compilation-environment'."
 (defvar counsel-compile-phony-pattern "^\\.PHONY:[\t ]+\\(.+\\)$"
   "Regexp for extracting phony targets from Makefiles.")
 
+(defvar counsel-compile-help-pattern
+  "\\(?:^\\(\\*?\\)[[:space:]]+\\([[:alnum:]]+\\)[[:space:]]+-\\)"
```

Comment:
The first group can match the empty string, which means the `(match-beginning 1)` in `counsel-compile--probe-make-help` will always return non-`nil`.

Reply:
Change `\\(\\*?\\)` to `\\(\\*\\)?` and then `(match-beginning 1)` will return `nil` when there is no asterisk.

Reply:
So I write my regex's in rx (in reb) so I had used (: bol (group (zero-or-one "*"))) for the initial part. I wonder what the rx equivalent would be to generate that?

Reply:
```el
(equal (rx bol (group (? ?*)) (+ space) (group (+ alnum)) (+ space) ?-)
       "^\\(\\*?\\)[[:space:]]+\\([[:alnum:]]+\\)[[:space:]]+-")

(equal (rx bol (? (group ?*)) (+ space) (group (+ alnum)) (+ space) ?-)
       "^\\(\\*\\)?[[:space:]]+\\([[:alnum:]]+\\)[[:space:]]+-")
```

See also https://elpa.gnu.org/packages/xr.html.

Reply:
@basil-conto awesome, thanks for the heads up